src/main/java/io/javalin/HttpResponseException.kt

```diff
@@ -8,19 +8,19 @@ package io.javalin
 
 import org.eclipse.jetty.http.HttpStatus
 
-open class HttpResponseException(val status: Int, @Deprecated("Use 'message' instead") val msg: String, val details: Map<String, String> = mapOf()) : RuntimeException(msg)
+open class HttpResponseException(val status: Int, @Deprecated("Use 'message' instead") val msg: String, val details: Map<String, String> = mapOf(), cause: Throwable? = null) : RuntimeException(msg, cause)
 
 class RedirectResponse(status: Int = HttpStatus.FOUND_302, msg: String = "Redirected") : HttpResponseException(status, msg)
 
-class BadRequestResponse(message: String = "Bad request") : HttpResponseException(HttpStatus.BAD_REQUEST_400, message)
+class BadRequestResponse(message: String = "Bad request", cause: Throwable? = null) : HttpResponseException(HttpStatus.BAD_REQUEST_400, message, cause = cause)
```

Comment:
Why `Throwable`?

Reply:
The ValidationException from the json Schema Validator contains useful information about what was not valid. With this the exception can be added to the Response and be left out or put in the error message sent by the server

Reply:
A Question from me: is it useful to use here also @JvmOverloads to keep compatibility to Java-Code using this constructors? Or is there another (more elegant) solution?

Reply:
`@JvmOverloads` is required, but isn't `Exception` better suited than `Throwable`? `Throwable` includes `Error` objects: 

> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch. Most such errors are abnormal conditions (https://docs.oracle.com/javase/6/docs/api/java/lang/Error.html)

Reply:
Is there any interest  in a PR that just adds the exception to the response? Jackson produces very useful output that would be nice to send down to the caller.
`
Ex. "com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException: nstantiation of [simple type, class com.me.test.foo$Bar] value failed for JSON property device_id due to missing (therefore NULL) value for creator parameter device_id which is a non-nullable type"`